enterprise/api_client/discovery.py

```diff
@@ -132,11 +142,22 @@ def get_catalog_results(self, content_filter_query, query_params=None, traverse_
         query_params = query_params or {}
 
         try:
-            endpoint = getattr(self.client, self.SEARCH_ALL_ENDPOINT)
-            response = endpoint().post(data=content_filter_query, **query_params)
-            if traverse_pagination:
-                response['results'] = self.traverse_pagination(response, endpoint, content_filter_query, query_params)
-                response['next'] = response['previous'] = None
+            cache_key = utils.get_cache_key(
+                service='discovery',
```

Comment:
nit: no need to pass `service` I think `endpoint` should be sufficient

Reply:
Actually, I added `service key` to make sure cache_key is unique, and will not conflict with any other cache_key.
because `endpoint` contains only `search/all/` that is a common endpoint name. so I was afraid It might conflict with other platform services, maybe in the future?